airflow/providers/apache/beam/operators/beam.py

```diff
@@ -520,12 +524,26 @@ class BeamRunGoPipelineOperator(BeamBasePipelineOperator):
         For more detail on Apache Beam have a look at the reference:
         https://beam.apache.org/documentation/
 
-    :param go_file: Reference to the Go Apache Beam pipeline e.g.,
-        /some/local/file/path/to/your/go/pipeline/file.go
+    :param go_file: Reference to the Apache Beam pipeline Go source file,
+        e.g. /local/path/to/main.go or gs://bucket/path/to/main.go.
+        Exactly one of go_file and go_binary must be provided.
+
+    :param go_binary: Reference to the Apache Beam pipeline Go binary compiled for the launching platform,
+        e.g. /local/path/to/launcher-main or gs://bucket/path/to/launcher-main.
+        Exactly one of go_file and go_binary must be provided.
+
+    :param worker_binary: Reference to the Apache Beam pipeline Go binary compiled for the worker platform,
+        e.g. /local/path/to/worker-main or gs://bucket/path/to/worker-main.
+        Needed if the OS or architecture of the workers running the pipeline is different from that
+        of the platform launching the pipeline. If not set, will default to the value of go_binary.
+        For more information, see the Apache Beam documentation for Go cross compilation:
+        https://beam.apache.org/documentation/sdks/go-cross-compilation/
```

Comment:
Need to mention this has no affect is `go_binary` is not set. Maybe an alternative would to make `go_binary` accept a `str | tuple[str, str]` and get rid of this parameter altogether?

Reply:
Thanks, I think it is clearer to have two different parameters, but agree the effect of specifying the launcher vs worker binary needs clarification. Updated in the latest commit. I also renamed the `go_binary` param to `launcher_binary` to be even more explicit.